### ISSUE TITLE
t.Date() type in Elysia.js results in Date & string in Eden, causing type mismatch

### ISSUE THREAD

**Issue by k3341095** [elysiajs/elysia#912]
### What version of Elysia is running?

1.1.24

### What platform is your computer?

ubuntu 22.04

### What steps can reproduce the bug?

Hello,

I'm currently using Elysia.js, and I’ve run into an issue with the t.Date() type. When I define a field with t.Date(), Eden interprets this type as Date & string, making it difficult to work with due to a type mismatch.

I need some guidance on how to handle this situation or if there’s a workaround to ensure compatibility between the types in Elysia.js and Eden. Any assistance on how to resolve this type discrepancy would be greatly appreciated.

Thank you!

### What is the expected behavior?

date | string

### What do you see instead?

Date & string

**Comment by tobias-edwards** [comment]
Indeed, defining a type with `t.Date()` returns the date as an ISO-8601-formatted string

```ts
const CreatedAt = t.Object({
  created_at: t.Date({
    description: 'Creation time'
  }),
})

// API
return created_at; // "2025-06-01T00:00:00.000Z"
```

When used with Eden, there is a type mismatch because a date is expected but a string is returned.

Currently, I'm having to manage two types for this field, one on `t.Date()` and one on `t.String()` and explicitly mapping `created_at` to a string before returning it.

**Comment by tobias-edwards** [comment]
After looking at the Elysia code, when `Date` is encoded it is mapped to an ISO string and then cast back to `TDate`:
https://github.com/elysiajs/elysia/blob/a35b26de4451da96f993917ca97cb95b5fc0401a/src/type-system.ts#L443

Since the logic is in place to convert this to a string I can augment the typing and things work as expected on my end:

```ts
// elysia.d.ts
import type { DateOptions, TString } from '@sinclair/typebox'

declare module '@sinclair/typebox' {
  interface JavaScriptTypeBuilder {
    Date(options?: DateOptions): TString
  }
}
```

Unfortunately, it doesn't seem as simple a fix for me to raise a PR here because amending to:

```ts
 .Encode((value) => value.toISOString()) as any as TString 
```

throws a type error in src/type-system.ts and I'm not sure how to fix this.

**Comment by joeh-nila** [comment]
Is there an interim solution for this? I get caught a lot of date de/serialisation

**Comment by crishoj** [comment]
For me, this now works with Elysia v1.4.
FYI @SaltyAom
